code/utils.py

```diff
@@ -1,12 +1,15 @@
 import json
 import pathlib
+from collections import Counter
 
 import contextily as cx
 import cv2
 import geopandas
+import matplotlib.gridspec as gridspec
```

Comment:
```suggestion
from matplotlib import gridspec
```

@jGaboardi any idea why Ruff did not complain about this? Normally it does? Maybe some rules to allow?

Reply:
I do not know... but I have noticed recently some instance in my work stuff where `ruff` is silent on things that it should be complaining about. Perhaps it's a configuration difference that has changed in the latest `ruff` releases.